sky/packages/sky/lib/animation/scroll_behavior.dart

```diff
@@ -39,20 +40,38 @@ class BoundedBehavior extends ScrollBehavior {
     return scrollOffset.clamp(minScrollOffset, maxScrollOffset);
   }
 
-  final double minScrollOffset = 0.0;
-  double get maxScrollOffset => math.max(0.0, _contentsExtents - _containerExtents);
+  double minScrollOffset = 0.0;
+  double get maxScrollOffset => math.max(minScrollOffset, minScrollOffset + _contentsExtents - _containerExtents);
 
   double applyCurve(double scrollOffset, double scrollDelta) {
     return (scrollOffset + scrollDelta).clamp(minScrollOffset, maxScrollOffset);
   }
 }
 
+class UnboundedBehavior extends BoundedBehavior {
```

Comment:
OK. How about an abstract ExtentScrollBehavior class; BoundedScrollBehavior and UnboundedScrollBehavior subclasses define what minScrollOffset maxScrollOffset mean.  Sound OK?